pkg/controller/atlasdeployment/searchindex.go

```diff
@@ -28,20 +28,19 @@ type searchIndexReconciler struct {
 
 func (sr *searchIndexReconciler) Reconcile(spec *akov2.SearchIndex, previous *status.DeploymentSearchIndexStatus) workflow.Result {
 	var stateInAtlas, stateInAKO *searchindex.SearchIndex
+	name := ""
```

Comment:
side note: if `previous == nil` then `reconcileInternal` will be invoked with empty string `""`. This needs to be reflected in the tests and asserted how the system should behave.